### PR TITLE
Handle underbar header when programatically scrolling

### PR DESCRIPTION
**Pull request by mcleinman**
**Fixes Phabricator ticket:** none, bug came from PR https://github.com/wikimedia/wikipedia-ios/pull/3677

### Notes
* A programatic scroll doesn't call the relevant delegate functions that allow `NavigationBarHider` to work. This adds `isProgramaticallyScrolling` to allow it to work in these case.

### Test Steps
1. Run app.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today'`. Ensure `On This Day` VC opens appropriately, with header looking good. Scroll up and down, nothing should jump.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today?0'`. Ensure `On This Day` VC opens appropriately, with header looking good. Scroll up and down, nothing should jump.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today?1'`. Ensure `On This Day` VC opens appropriately, with header looking good. Scroll up and down, nothing should jump.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today?6'`. Ensure `On This Day` VC opens appropriately, with header looking good. Scroll up and down, nothing should jump.

### CODE REVIEW

File: Wikipedia/Code/NavigationBarHider.swift [hunk 1]
```diff
@@ -32,23 +32,43 @@ public class NavigationBarHider: NSObject {
...
+    private func setDataForScroll(yOffset: CGFloat, topContentInset: CGFloat) {
+        guard let navigationBar = navigationBar else {
+            return
+        }
+        initialScrollY = yOffset + topContentInset
+        initialNavigationBarPercentHidden = navigationBar.navigationBarPercentHidden
+    }
+
+    public func setIsProgramaticallyScrolling(_ isProgramaticallyScrolling: Bool, yOffset: CGFloat, topContentInset: CGFloat) {
+        self.isProgramaticallyScrolling = isProgramaticallyScrolling
```
Comment (tonisevener) [inline Wikipedia/Code/NavigationBarHider.swift:48]: Might want a `guard let isInteractiveHidingEnabled == true` here, just to confirm that the user has set the right navbar settings before you update these internal state properties.

File: Wikipedia/Code/NavigationBarHider.swift [hunk 2]
```diff
@@ -32,23 +32,43 @@ public class NavigationBarHider: NSObject {
...
     }
 
     @objc public func scrollViewDidScroll(_ scrollView: UIScrollView) {
         guard let navigationBar = navigationBar else {
             return
         }
-        
+
+        /// When programatically scrolling on a just-created scroll view, we need a layout pass to get accurate heights.
+        scrollView.layoutIfNeeded()
```
Comment (tonisevener) [inline Wikipedia/Code/NavigationBarHider.swift:69]: If it only seems needed when programmatically scrolling, might want to wrap these in `if isProgramaticallyScrolling {` to limit unneeded layout passes in other states.

**Review by tonisevener — CHANGES_REQUESTED** [review]
Just a couple of small suggestions. Having trouble installing the widgets though to test (Widgets extension is now missing, but nothing changed in the project file in this PR diff, so I'm very confused!)

File: Wikipedia/Code/NavigationBarHider.swift (hunk 1 above)
Comment (mcleinman) [reply Wikipedia/Code/NavigationBarHider.swift:52]: Good call, updated!

File: Wikipedia/Code/NavigationBarHider.swift (hunk 2 above)
Comment (mcleinman) [reply Wikipedia/Code/NavigationBarHider.swift]: Good call, updated!

**Review by tonisevener — APPROVED** [review]

**Pull request merged**